### PR TITLE
T-SEC-5: Reject non-same-origin proxy mutations

### PR DESCRIPTION
**Pull request by manumissio**
## What changed

- Add a shared same-origin guard before frontend proxy POSTs read or forward the server-side API key.
- Reject `same-site` and `cross-site` Fetch Metadata, mismatched or malformed `Origin`, invalid `Host`/`X-Forwarded-Proto` metadata, explicit-port mismatches, and POST method wiring mismatches.
- Preserve exact same-origin requests and headerless non-browser callers.
- Pass the incoming request through all five mutation routes and document the reverse-proxy trust contract in `SECURITY.md`.
- Add tests for the shared boundary, every POST route, Next standalone URL behavior, explicit ports, malformed authorities, and existing forwarding/configuration behavior.

## Why

The frontend proxy injects `API_AUTH_KEY`, which authenticates the frontend deployment rather than the visitor. Without an origin check, a hostile site could cause a browser to submit mutation requests through that trusted frontend boundary.

## Trust-boundary impact

This changes the Internet-to-Frontend boundary documented in `SECURITY.md`. Cross-site browser requests lose access to proxy mutation routes before the API key is read. Reverse proxies must preserve the public `Host` and overwrite incoming `X-Forwarded-Proto`; `X-Forwarded-Host` is deliberately ignored. No secret, permission, application API, runtime default, or person-data behavior changes.

## Risk and compatibility

Headerless non-browser POSTs remain allowed for existing server-side callers. That is an explicit compatibility exception and is not complete protection for legacy browsers that omit both `Origin` and Fetch Metadata.

## Verification

- PASS: `cd frontend && npm test` (27 tests)
- PASS: `cd frontend && npm run build -- --webpack`
- PASS: production standalone smoke for direct, reverse-proxy, explicit-port, port-mismatch, cross-site, protocol-injection, and malformed-Host cases
- PASS: `PYTHONPATH=. .venv/bin/pytest -q tests/test_frontend_pages_config.py tests/test_resultcard_agenda_status_refresh.py tests/test_search_sort_ui_guardrails.py tests/test_semantic_search_ui_guardrails.py` (14 tests)
- PASS: `PYTHONPATH=. .venv/bin/pytest -q tests/test_docs_links.py` (2 tests)
- PASS: `PYTHONPATH=. .venv/bin/pytest -q` (1,190 tests)
- PASS: `git diff --check origin/master`
- PASS: independent pre-commit review, no remaining P1/P2 findings

## Remaining work

T-SEC-5 stays in progress until this PR merges and an evidence-only ledger closure records the merged checks. G2 still gates T-SEC-4, and G3 still gates Phase 2 architecture work.

### CODE REVIEW

**Comment by manumissio** [comment]
@codex review

File: frontend/app/api/_lib/backend.js [hunk 1]
```diff
@@ -12,6 +21,96 @@ function getApiAuthKey() {
...
+      : parseHostAuthority(hostHeader, protocol);
+  if (host === null) {
+    return null;
+  }
+  return parseWebOrigin(`${protocol}://${host}`);
+}
+
+function isNonSameOriginMutation(request, method) {
+  if (method !== MUTATION_METHOD && request?.method !== MUTATION_METHOD) {
+    return false;
```
[the remark this answers is not in the document]
Comment (manumissio) [reply frontend/app/api/_lib/backend.js:99]: Fixed in e951f9d. The guard now rejects any supplied request/proxy method disagreement before origin evaluation. New tests cover both POST-to-GET and GET-to-POST mismatches without Origin or Fetch Metadata; the targeted suite passes 14/14, the frontend suite 27/27, the production build/smoke pass, and the full Python suite passes 1,190/1,190. A fresh independent review found no remaining P1/P2 issues.

**Comment by manumissio** [comment]
@codex review
